lib/Cake/Test/Case/Network/CakeRequestTest.php

```diff
@@ -215,9 +215,14 @@ public function testQueryStringAndNamedParams() {
 		$request = new CakeRequest();
 		$this->assertEquals('some/path', $request->url);
 
-		$_SERVER['REQUEST_URI'] = Configure::read('App.fullBaseUrl') . '/other/path?url=https://cakephp.org';
+		$base = Configure::read('App.fullBaseUrl');
+		$_SERVER['REQUEST_URI'] = $base . '/other/path?url=https://cakephp.org';
 		$request = new CakeRequest();
 		$this->assertEquals('other/path', $request->url);
+
+		$_SERVER['REQUEST_URI'] =  str_repeat('x', strlen($base) - 4) . '://?/other/path';
+		$request = new CakeRequest();
+		$this->assertEquals('', $request->url);
```

Comment:
Just reading the code, why does this not equal `/`?
```
if (empty($uri) || $uri === '/' || $uri === '//' || $uri === '/index.php') {
	$uri = '/';
}
```

Reply:
Not entirely sure, but I was aiming to make the minimum amount of changes to resolve the problem.